lib/postzord/receiver/local_batch.rb

```diff
@@ -25,6 +25,15 @@ def receive!
     # 09/27/11 this is slow
     notify_users
 
+    if @object.respond_to?(:type) && @object.type == "Reshare"
+      root_object = Post.find_by guid: @object.root_guid
```

Comment:
I'd guess `@object.root` would work and probably be already loaded. But I didn't verify it and threw a debugger into that section just to check. Since you developed this code, you might have a good reason I'm overlooking, so I ask.